libr/util/idpool.c

```diff
@@ -163,6 +162,71 @@ R_API void* r_id_storage_get(RIDStorage* storage, ut32 id) {
 	return storage->data[id];
 }
 
+R_API bool r_id_storage_get_lowest(RIDStorage *storage, ut32 *id) {
+	r_return_val_if_fail (storage, false);
+	int fd = 0;
+	if (!r_id_storage_get_prev (storage, (ut32 *)&fd) && fd == 0) {
```

Comment:
yep, 0 is a valid id. id_storage has no ids indicating errors. returnvalue indicates errors

Reply:
this is used to get the highest and lowest fds, fd is set to 0 and then calls to get the next and the prev. to find the first and last of them. so in those cases fd == 0 means that there's no lower or higher fds, but also, there's no next or prev (otherwise that would be reported in the fd !=0)

Reply:
im not 100% happy about this api design, but it works for me

Reply:
what you want to check is if the storage is empty, right?

Reply:
id_next and id_prev do not require a valid id as parameter, except the id should be in the possible range of the idstorage.
if the id is not valid, but in that range, prev and next are intended to provide the next/previous valid id, as in a ringbuffer.

Reply:
In any case, I don't think `fd == 0` is right. If there is no "lowest" you should get `false` from `r_id_storage_get_prev` and that is a good enough error condition to check.

Reply:
yes i know this is a hack because i was trying to use that idstorage api, without changing it a lot. thats why i didnt liked the API at first and wanted to have it merged for testing with my PR. for me i was happy to get the functionality, but the api was duplicating the same logic in multiple placs.. i mean. iofd an riodesc are just frontends for idpool which end up just giving a different api usage,